File: android/src/main/kotlin/dev/jonathanvegasp/result_channel/Writer.kt
package dev.jonathanvegasp.result_channel

interface Writer {
    fun byte(value: Byte)

    fun char(value: Char)

    fun int(value: Int)

    fun long(value: Long)

    fun float(value: Float)

    fun double(value: Double)

    fun byteArray(value: ByteArray)

    fun intArray(value: IntArray)

    fun longArray(value: LongArray)

    fun floatArray(value: FloatArray)

    fun doubleArray(value: DoubleArray)

    fun toByteArray(): ByteArray
}

File: ios/Classes/Writer.swift
import Foundation

public protocol Writer {
    func byte(value: UInt8)
    func char(value: UInt16)
    func int(value: Int32)
    func long(value: Int64)
    func float(value: Float)
    func double(value: Double)
    func byteArray(value: Data)
    func intArray(value: [Int32])
    func longArray(value: [Int64])
    func floatArray(value: [Float])
    func doubleArray(value: [Double])
    func toByteArray() -> Data
}
